### PR TITLE
Dollification Admin Smite

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request

Adds a dollification admin smite. It's not instant, instead it has four stages of "infection" and it's curable via 20u Dinitrogen Plasmide. At stage 2 it gives heels. Stage 3 is a gag. Stage 4 replaces your name to your job title with a random number (Yes, on the manifest too), and prompts you if you want it permanent. 

- Checks for a new pref: transformation. 
- OOC Escape / Safeword deletes it entirely.
- Health analyzer tells you the cure.

## Why It's Good For The Game

This is the basic component and I want to expand on it later with a very funny idea I had in mind with job specific gear. But for now admins can get a funny smite since this is the primary component.

Also it's hot.

## Proof Of Testing

- [x] Properly deletes itself. Reverting everything.
- [x] Properly handles its logic and progresses properly
- [x] Is curable, with the option for it not to be.
- [x] Smite works.

image

## Changelog

:cl:
add: Dollification Admin Smite
add: Transformation ERP preference. 
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
god is dead

**Comment by StrangeWeirdKitten** [comment]
> god is dead

has been

**Comment by BurgerLUA** [comment]
image

**Comment by BurgerLUA** [comment]
Make the cure mannitol tbh

**Comment by ReturnToZender** [comment]
I'm gonna have to bring back femboyification at this rate

File: modular_zubbers/code/datums/components/dollification.dm [hunk 1]
```diff
@@ -0,0 +1,200 @@
+/// Item handling
+
+/obj/item/clothing/mask/muzzle/ring/doll
+	name = "fused opened lips"
+	desc = "Your lips are stuck open like this. No matter how much you prod at them, you can't close them!"
+
+/obj/item/clothing/shoes/fancy_heels/doll
+	name = "attatched high heels"
+	desc = "These obsidian heels refuse to detatch from your feet. They feel a part of you now."
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm:9]: They feel like a part of you now.

File: modular_zubbers/code/datums/components/dollification.dm [hunk 2]
```diff
@@ -0,0 +1,200 @@
+/// Item handling
+
+/obj/item/clothing/mask/muzzle/ring/doll
+	name = "fused opened lips"
+	desc = "Your lips are stuck open like this. No matter how much you prod at them, you can't close them!"
+
+/obj/item/clothing/shoes/fancy_heels/doll
+	name = "attatched high heels"
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm]: Attached

File: modular_zubbers/code/datums/components/dollification.dm [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+/datum/smite/dollify/effect(client/user, mob/living/target)
+	. = ..()
+
+	if(!iscarbon(target))
+		to_chat(user, span_warning("This must be used on a carbon mob."), confidential = TRUE)
+		return
+
+	if(!target.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy)) // Another pref check.
+		to_chat(user, span_warning("ERP sex toy prefrences not enabled."), confidential = TRUE)
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm:24]: preferences

File: modular_zubbers/code/datums/components/dollification.dm [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+	origin = pick(orgins)
+	to_chat(doll, span_purple("<b>You feel something slick smudge across your [origin]; a glossy smear visibly spreading in every direction!</b>"))
+	progress_stage()
+	START_PROCESSING(SSobj, src)
+
+/datum/component/dollification/Destroy()
+	QDEL_NULL(gag)
+	QDEL_NULL(heels)
+	UnregisterSignal(doll, COMSIG_ATOM_EXAMINE)
+	to_chat(doll, span_alert("You feel the rubbery changes hault in their tracks before reverting you back to normal."))
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm:83]: halt*

File: modular_zubbers/code/datums/components/dollification.dm [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+		stage++
+
+	switch(stage)
+		if(1)
+			examine_text = "There is a festing splotch of smoothness slowly encompassing their [origin]"
+			infection_text = list(
+							"You feel the elastic oozing into your [origin]. Into the muscle itself.",
+							"Your [origin] is tingling as the slick, cool smear fully encompasses it.",
+							"It's feeling tight around your [origin], as a million lines of tension slowly sculpt away imperfection",
+							"A pleasent tingling crawls through your body.")
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm:143]: pleasant

File: modular_zubbers/code/datums/components/dollification.dm [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+		if(2)
+			to_chat(doll, span_purple("<b>Wait are those heels? Sheets of obsidian black spontaniously wrap around and attatch to both your feet.</b>"))
+			examine_text = "The rubbery sheen continues to spread, overtaking more and more of their body."
+			infection_text = list(
+							"You're becoming more rubbery as time goes on and it's feeling really good.",
```
Comment (SovietSpaghetti) [inline modular_zubbers/code/datums/components/dollification.dm:148]: maybe a comma after on

**Review by SovietSpaghetti — COMMENTED** [review]
That should be all the SPaG mistakes I saw

**Comment by CydiaButt13** [comment]
This is fantastic. I suggest more clothing to go along with it, fused latex maid outfit comes to mind, something to cover your head too perhaps?

**Comment by NanoCats** [comment]
> This is fantastic. I suggest more clothing to go along with it, fused latex maid outfit comes to mind, something to cover your head too perhaps?

Personally I think there should be some choices along the way that shape what kind of doll you become.

**Comment by SovietSpaghetti** [comment]
I do feel like it's weird to have specifically as a smite though, for reasons other people have brought up outside of the PR, mainly how it wouldn't solely be limited to a scene tool. Maybe a lustwish item or something, with a pref tied to it. Maybe different variants available at the vendor for the things other people have brought up.

**Comment by projectkepler-ru** [comment]
https://www.youtube.com/watch?v=f9HCzAXoDI8

**Comment by Majkl-J** [comment]
> Edit: Also probably make sure this can't roll from a 1 on the die of fate.

Oh god I forgot that was a thing. But tbh from what I heard shark mostly made it a smite for the sake of testing it.

**Comment by NanoCats** [comment]
> I do feel like it's weird to have specifically as a smite though, for reasons other people have brought up outside of the PR, mainly how it wouldn't solely be limited to a scene tool. Maybe a lustwish item or something, with a pref tied to it. Maybe different variants available at the vendor for the things other people have brought up.
> 
> Edit: Also probably make sure this can't roll from a 1 on the die of fate.

Add TF preference, make it so it can roll on a 1 from the die of fate if you have it enabled :wao:

**Comment by StrangeWeirdKitten** [comment]
> I do feel like it's weird to have specifically as a smite though, for reasons other people have brought up outside of the PR, mainly how it wouldn't solely be limited to a scene tool. Maybe a lustwish item or something, with a pref tied to it. Maybe different variants available at the vendor for the things other people have brought up.
> 
> Edit: Also probably make sure this can't roll from a 1 on the die of fate.

Die of fate does not randomize admin smites. It has its own random switch logic tree that chooses.

> > I do feel like it's weird to have specifically as a smite though, for reasons other people have brought up outside of the PR, mainly how it wouldn't solely be limited to a scene tool. Maybe a lustwish item or something, with a pref tied to it. Maybe different variants available at the vendor for the things other people have brought up.

> > Edit: Also probably make sure this can't roll from a 1 on the die of fate.
> 
> Add TF preference, make it so it can roll on a 1 from the die of fate if you have it enabled :wao:

TF pref created.

**Comment by Hislyna** [comment]
This should just be a lustwish item with the same effect instead of a smite imo

**Comment by StrangeWeirdKitten** [comment]
> This should just be a lustwish item with the same effect instead of a smite imo

It's a draft for a reason

**Comment by aKromatopzia** [comment]
> I'm gonna have to bring back femboyification at this rate

not the femboy vulpakin smite

**Comment by funnyfreeperson** [comment]
if there could be a less latex focused version of this, that would be awesome- like onahole or more traditional doll

**Comment by NanoCats** [comment]
> if there could be a less latex focused version of this, that would be awesome- like onahole or more traditional doll

I mean at that rate I feel like you could just put someone's soul right into one, I think that exists already.

**Comment by funnyfreeperson** [comment]
> > if there could be a less latex focused version of this, that would be awesome- like onahole or more traditional doll
> 
> I mean at that rate I feel like you could just put someone's soul right into one, I think that exists already.

kinda, but not really, it's not quite the same, and the items don't exist.

**Comment by LordSaladin** [comment]
;3

**Pull request closed without merging**